### DISCUSSION TITLE
vue-socket.io - cannot use socket in App

### DISCUSSION THREAD

**Discussion by mzwach** [quasarframework/quasar#15312] [General - Components / Directives / etc]
I am new to Quasar Framework and cannot figure out the problem, trying to initialize cue-socket.io in a boot file.

When I try to use the boot class, I cannot pass my socket connection with
app.use( SocketInstance );
to my App.vue in order to get the methods within sockets: section working

Quasar console logging error says:
TypeError: undefined is not an object (evaluating 't.prototype.$socket=this.io')

In case that I do not try to call the app.use function, Socket connection seems to be established,
but I have no access via the this.$socket for e.g. calling the this.$socket.emit function.

Here is my code:
App.vue

```
<template>
  <router-view />
</template>

<script>
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'App',

  data() {
     return {
        //socket: {},
        //socket: app.config.globalProperties.$socket,
        //socket: socket,
        axios: {},
        is_connected: false,
        socketMessage: false,
     }
  },

  sockets: {
    connect() {
      // Fired when the socket connects. ... is NEVER fired!
      this.isConnected = true;
      console.log('socket connected');
    },

    disconnect() {
      this.isConnected = false;
      
      console.log('socket disconnected');
    },

    // Fired when the server sends something on the "messageChannel" channel.
    messageChannel(data) {
      this.socketMessage = data
      console.log('messageChannel',data);
    },

  },
  mounted() {

     //this.socket=app.config.globalProperties.$socket;
     console.log('mounted')

     
    // this.$socket.emit('data.get', {action: 'get'}, (response) => {
    //        console.log('einsatz.get status:' + response.status);
    //   })


  },
  methods: {
 

  },

})
</script>
```

**boot section in quasar.config.js**
```
boot: [
      'my-vue-socket-io', // added to boot socket-io, create with: quasar new boot socket-io
    ],
```

**and finally my boot/my-vue-socket-io.js**

```
import { boot } from 'quasar/wrappers'
import Vue from 'vue'
//import socketio from 'socket.io';  //use vue-sockeet.io instead
import SocketIO from 'socket.io-client';
import VueSocketIO from 'vue-socket.io';
import store from 'boot/store';
//import { io } from 'socket.io-client';

const socket_url = 'https://test.xxx.com';
localStorage.jwt = 'xxxxx';
localStorage.username='xxx';



export default boot(async({ app }) => {

  const SocketInstance = new VueSocketIO({
        debug: true,
        connection: socket_url+'?jwt='+localStorage.jwt+'&user='+localStorage.username,
        vuex: {
          //store,
          actionPrefix: 'SOCKET_',
          mutationPrefix: "SOCKET_"
        },
        options: { path: "/socket.io/" } //Optional options
    })

  // giving me the error !!!!
  app.use( SocketInstance );

  app.config.globalProperties.$socket = SocketInstance
 
})
```

I would really appreciate, whether there's a problem in compatibility with the vue-socket.io library or with my code or with the framework!

Thank you in advance,
Regards,
Michael

**Comment by suleimanas** [comment]
shouldn't be: `Vue.use` instead of _app.use_

**Reply by mzwach** [reply]
I have tried `Vue.use` , but Vue was not within scope (undefined) and also not valid within the Object Array of boot().

**Comment by mzwach** [comment]
Has someone else an idea or a code example?
Or is it better to use another socket.io library instead?

**Comment by yusufkandemir** [comment]
The package you are using is [`vue-socket.io`](https://www.npmjs.com/package/vue-socket.io) which is released 3 years ago and does not have Vue 3 support. You need to find a more up-to-date binding that supports Vue 3. I'd say unless you want something like Vuex/Pinia integration, you should be good to go with the plain client(e.g. `socket.io-client`) and not a Vue-specific binding.
